code/Helper/Entity/Categoryhelper.php

```diff
@@ -160,8 +157,16 @@ public function getObject(Mage_Catalog_Model_Category $category)
             'product_count'   => $category->getProductCount(),
         ];
 
-        if (!empty($image_url)) {
-            $data['image_url'] = $image_url;
+        try {
+            $imageUrl = $this->getThumbnailUrl($category) ?: $category->getImageUrl();
```

Comment:
cc @Inwerpsel
Makes sense to you? Indexing thumbnail when it's available. Otherwise index main image.